engine/grid.py

```diff
@@ -248,31 +282,38 @@ def area(p1, p2, p3):
         # is same as A
         return A == (A1 + A2 + A3)
 
-    def activate_line(self, coordinate, n, is_horizontal):
+    def activate_line(self, coordinate, n, is_vertical):
         """
         Activate a single horizontal line of length n starting at row, col.
         If is_horizontal is True, the line is from (row, col) to (row, col + n).
         Otherwise, the line is from (row, col) to (row + n, col).
         Args:
-            coordinate (int tuple): coordinate of the starting node         
+            coordinate (int tuple): coordinate of the starting node
         """
         (row, col) = coordinate
         # Note: rows are y-position and columns are x-position
-        if is_horizontal:
+        if not is_vertical:
             for i in range(n):
-                self.nodes[col+i][row].is_active = True
+                self.nodes[col + i][row].is_active = True
+                self.horizontal_waypoint_list.append([self.nodes[col + i, row]])
         else:
             for i in range(n):
-                self.nodes[col][row+i].is_active = True
+                self.nodes[col][row + i].is_active = True
+                self.vertical_waypoint_list.append([self.nodes[col, row + i]])
 
     def activate_triangle(self, p1, p2, p3):
         """
         Activates all the nodes in a triangle based on three vertices p1, p2, p3.
         """
+        self.horizontal_waypoint_list = [[] for _ in range(self.num_cols)]
```

Comment:
can we initialize this in the class constructor and only need to change it for activate_rectangle?

Reply:
This is also needed for shapes like circle or any slanted shapes we may have

Reply:
yeah, but activate_circle uses that exact same initialization

Reply:
Good point